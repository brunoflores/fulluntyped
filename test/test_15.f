Counter = {get: Unit -> Nat, inc: Unit -> Unit};
inc3 = lambda x: Counter. (x.inc unit; x.inc unit; x.inc unit);

c = let x = ref 1 in
      { get = (lambda _:Unit. !x),
        inc = (lambda _:Unit. x := succ (!x)) };

(inc3 c; c.get unit);
