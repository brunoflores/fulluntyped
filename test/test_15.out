Counter: { get: Unit->Nat, inc: Unit->Unit }
inc3: Counter -> Unit
c: { get: Unit->Nat, inc: Unit->Unit }
4: Nat
